src/engine.cpp

```diff
@@ -734,15 +734,45 @@ static void AcceptEnginePreview(EngineID eid, CompanyID company)
 		c->avail_roadtypes = AddDateIntroducedRoadTypes(c->avail_roadtypes | GetRoadTypeInfo(e->u.road.roadtype)->introduces_roadtypes, _date);
 	}
 
-	e->preview_company = INVALID_COMPANY;
-	e->preview_asked = (CompanyMask)-1;
 	if (company == _local_company) {
 		AddRemoveEngineFromAutoreplaceAndBuildWindows(e->type);
+
+		/* Update the toolbar. */
+		InvalidateWindowData(WC_MAIN_TOOLBAR, 0);
+		if (e->type == VEH_ROAD) InvalidateWindowData(WC_BUILD_TOOLBAR, TRANSPORT_ROAD);
+		if (e->type == VEH_SHIP) InvalidateWindowData(WC_BUILD_TOOLBAR, TRANSPORT_WATER);
```

Comment:
why only road & ship?

Reply:
Can you test this to see if it works?
```suggestion
		if (e->type == VEH_SHIP) InvalidateWindowData(WC_BUILD_TOOLBAR, TRANSPORT_WATER);
		if (e->type == VEH_TRAIN) InvalidateWindowData(WC_BUILD_TOOLBAR, TRANSPORT_RAIL);
		if (e->type == VEH_AIRCRAFT) InvalidateWindowData(WC_BUILD_TOOLBAR, TRANSPORT_AIR);
```

Reply:
Because only ship and road windows can be opened when no vehicles are available. Or so it was before NRT, now it's ship window only. I can remove the update for the road window if you want but IMO it should be fixed on NRT side since building roads kind of makes sense even without vehicles (for towns).